extensions/amp-ad-network-sulvo-impl/0.1/amp-ad-network-sulvo-impl.js

```diff
@@ -0,0 +1,2016 @@
+  googleBlockParameters,
+  googlePageParameters,
+  groupAmpAdsByType,
+  isCdnProxy,
+  isReportingEnabled,
+  maybeAppendErrorParameter,
+  truncAndTimeUrl,
+} from '../../../ads/google/a4a/utils';
+import {CONSENT_POLICY_STATE} from '../../../src/consent-state';
+import {Deferred} from '../../../src/utils/promise';
+import {
+  FlexibleAdSlotDataTypeDef,
+  getFlexibleAdSlotData,
+} from './flexible-ad-slot-utils';
+import {Layout, isLayoutSizeDefined} from '../../../src/layout';
+import {Navigation} from '../../../src/service/navigation';
+import {RTC_VENDORS} from '../../../src/service/real-time-config/callout-vendors';
+import {
+  RefreshManager, // eslint-disable-line no-unused-vars
+  getRefreshManager,
+} from '../../amp-a4a/0.1/refresh-manager';
+import {SafeframeHostApi} from './safeframe-host';
+import {Services} from '../../../src/services';
+import {
+  TFCD,
+  constructSRABlockParameters,
+  serializeTargeting,
+  sraBlockCallbackHandler,
+} from './sra-utils';
+import {WindowInterface} from '../../../src/window-interface';
+import {
+  addAmpExperimentIdToElement,
+  extractUrlExperimentId,
+  isInManualExperiment,
+} from '../../../ads/google/a4a/traffic-experiments';
+import {
+  assertDoesNotContainDisplay,
+  setImportantStyles,
+  setStyles,
+} from '../../../src/style';
+import {
+  createElementWithAttributes,
+  isRTL,
+  removeElement,
+} from '../../../src/dom';
+import {deepMerge, dict} from '../../../src/utils/object';
+import {dev, devAssert, user} from '../../../src/log';
+import {domFingerprintPlain} from '../../../src/utils/dom-fingerprint';
+import {escapeCssSelectorIdent} from '../../../src/css';
+import {
+  getAmpAdRenderOutsideViewport,
+  incrementLoadingAds,
+  is3pThrottled,
+  waitFor3pThrottle,
+} from '../../amp-ad/0.1/concurrent-load';
+import {getCryptoRandomBytesArray, utf8Decode} from '../../../src/utils/bytes';
+import {getMode} from '../../../src/mode';
+import {getMultiSizeDimensions} from '../../../ads/google/utils';
+import {getOrCreateAdCid} from '../../../src/ad-cid';
+import {
+  isExperimentOn,
+  randomlySelectUnsetExperiments,
+} from '../../../src/experiments';
+
+import {insertAnalyticsElement} from '../../../src/extension-analytics';
+import {isArray} from '../../../src/types';
+import {isCancellation} from '../../../src/error';
+import {
+  lineDelimitedStreamer,
+  metaJsonCreativeGrouper,
+} from '../../../ads/google/a4a/line-delimited-response-handler';
+import {parseQueryString} from '../../../src/url';
+import {stringHash32} from '../../../src/string';
+import {tryParseJson} from '../../../src/json';
+
+/** @type {string} */
+const TAG = 'amp-ad-network-sulvo-impl';
+
+/** @const {string} */
+const DOUBLECLICK_BASE_URL =
+  'https://securepubads.g.doubleclick.net/gampad/ads';
+
+/** @const {string} */
+const RTC_SUCCESS = '2';
+
+/** @const {string} */
+const DOUBLECLICK_SRA_EXP = 'doubleclickSraExp';
+
+/** @const @enum{string} */
+const DOUBLECLICK_SRA_EXP_BRANCHES = {
```

Comment:
I'm not sure I understand the purpose of this implementation. Is this just a clone of `amp-ad-network-doubleclick-impl`? If so I think these will quickly go out of sync.